packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-actions/hooks/useActivateLastPublishedVersionWorkflowSingleRecordAction.ts

```diff
@@ -13,7 +13,8 @@ export const useActivateLastPublishedVersionWorkflowSingleRecordAction: SingleRe
       isDefined(workflowWithCurrentVersion) &&
       isDefined(workflowWithCurrentVersion.currentVersion.trigger) &&
       isDefined(workflowWithCurrentVersion.lastPublishedVersionId) &&
-      workflowWithCurrentVersion.currentVersion.status !== 'ACTIVE' &&
+      workflowWithCurrentVersion.lastPublishedVersionId !== '' &&
+      !workflowWithCurrentVersion.statuses?.includes('ACTIVE') &&
```

Comment:
I would rather do `isDefined(workflowWithCurrentVersion.lastPublishedVersionId) && workflowWithCurrentVersion.currentVersion.id !== workflowWithCurrentVersion.lastPublishedVersionId`. This way you don't have to handle statuses optimistic rendering

Reply:
It doesn't work, because id you deactivate a workflow, you cannot activate it again because the current version id will be the lastPublishedVersionId